app/js/pages/OffersList/index.jsx

```diff
@@ -50,6 +55,34 @@ class OffersList extends Component {
     this.setState({sortType});
   };
 
+  setLocation = (location) => {
+    if (!location) {
+      return this.setState({calculatingLocation: false, locationCoords: null});
+    }
+    if (location === this.state.location) {
+      return;
+    }
+
+    this.setState({calculatingLocation: true});
+    getLocation(location).then(coords => {
+      this.setState({
+        calculatingLocation: false,
+        locationCoords: coords,
+        location
+      });
+    }).catch(e => {
+      this.setState({
+        calculatingLocation: false,
+        error: e.message,
+        location
+      });
+    });
+  };
+
+  calculateDistance = (userCoords) => {
+    return  Math.sqrt(Math.pow(userCoords.lat - this.state.locationCoords.lat, 2) + Math.pow(userCoords.lng - this.state.locationCoords.lng, 2));
```

Comment:
👍 for the formula
👎 for the extra space between return and Math

Also should it be in utils?